applications/serial_lte_modem/doc/TCPIP_AT_commands.rst

```diff
@@ -0,0 +1,1541 @@
+
+  * 1: SOCK_STREAM for TCP
+  * 2: SOCK_DGRAM for UDP
+
+  When the ``<sec_tag>`` is specified:
+
+  * 1: SOCK_STREAM for TLS
+  * 2: SOCK_DGRAM for DTLS
+
+* The ``<role>`` parameter can accept one the following values:
+
+  * ``0`` - Client
+  * ``1`` - Server
+
+Response syntax
+~~~~~~~~~~~~~~~
+
+::
+
+   #XSOCKET: <handle>[,<type>,<protocol>]
+
+* The ``<handle>`` value is an integer and it can be interpreted as follows:
+
+  * Positive - The socket opened successfully.
+  * Negative - The socket failed to open.
+  * ``0`` - The socket closed successfully.
+
+* The ``<type>`` parameter value depends on the presence of the <sec_tag> parameter.
+  When the ``<sec_tag>`` is not specified:
+
+  * 1: SOCK_STREAM for TCP
+  * 2: SOCK_DGRAM for UDP
+
+  When the ``<sec_tag>`` is specified:
+
+  * 1: SOCK_STREAM for TLS
+  * 2: SOCK_DGRAM for DTLS
+
+* The ``<protocol>`` value is present only in the response to a request to open the socket.
+  It can be one of the following:
+
+  * IPPROTO_TCP(6)
+  * IPPROTO_UDP(17)
+  * IPPROTO_TLS_1_2(258)
+  * IPPROTO_DTLS_1_2(273)
+
+Unsolicited notification
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+::
+
+   #XSOCKET: <error> closed
+
+The ``<error>`` value is a negative integer, and it represents the error value according to the standard POSIX errorno.
+
+Examples
+~~~~~~~~
+
+::
+
+   AT#XSOCKET=1,1,0
+   #XSOCKET: 3, 6, 0
+   OK
+   AT#XSOCKET=1,2,0
+   #XSOCKET: 3, 17, 0
+   OK
+   AT#XSOCKET=0
+   #XSOCKET: 0, closed
+   OK
+   at#xsocket=1,1,0,16842755
+   #XSOCKET: 2, 1, 0, 258
+   OK
+   at#xsocket=1,2,0,16842755
+   #XSOCKET: 2, 2, 0, 273
+   OK
+
+Read command
+------------
+
+The read command allows you to check the socket handle.
+
+Syntax
+~~~~~~
+
+::
+
+   #XSOCKET?
+
+Response syntax
+~~~~~~~~~~~~~~~
+
+::
+
+   #XSOCKET: <handle>[,<protocol>, <role>]
+
+* The ``<handle>`` value is an integer and it can be interpreted as follows:
+
+  * Positive - The socket is valid.
+  * ``0`` - The socket is closed.
+
+* The ``<protocol>`` value is present only in the response to a request to open the socket.
+  It can be one of the following:
+
+  * IPPROTO_TCP(6)
+  * IPPROTO_UDP(17)
+  * IPPROTO_TLS_1_2(258)
+  * IPPROTO_DTLS_1_2(273)
+
+* The ``<role>`` parameter can accept one the following values:
+
+  * ``0`` - Client
+  * ``1`` - Server
+
+Examples
+~~~~~~~~
+
+::
+
+   AT#XSOCKET?
+   #XSOCKET: 3, 6, 0
+   OK
+
+::
+
+   AT#XSOCKET?
+   #XSOCKET: 3, 17, 0
+   OK
+
+::
+
+   at#xsocket?
+   #XSOCKET: 2, 258, 0
+   OK
+
+::
+
+   at#xsocket?
+   #XSOCKET: 2, 273, 0
+   OK
+
+Test command
+------------
+
+The test command tests the existence of the command and provides information about the type of its subparameters.
+
+Syntax
+~~~~~~
+
+::
+
+   #XSOCKET=?
+
+Response syntax
+~~~~~~~~~~~~~~~
+
+::
+
+   #XSOCKET: <list of op value>,<list of type value>,<list of roles>,<sec-tag>
+
+* The ``<role>`` parameter can accept one the following values:
+
+  * ``0`` - Client
+  * ``1`` - Server
```

Comment:
Is this description relevant also for the test command?

Reply:
Yes, both TCP and UDP proxy need to be assigned as Server or Client role explicitly